ROS2: Redesign motorized Joint

alek-kam-robotec-ai opened the pull request:
Work towards https://github.com/o3de/o3de-extras/issues/126.
This pr includes:
 - Decoupling of the Motorized Joint into separate Joint Motor Controllers.
 - Removing of test logic and implementing debug ImGui functionality.
 - Transition from the use of Rigid body impulses to the PhysX Joint Request Bus.

Signed-off-by: Aleksander Kamiński <aleksander.kaminski@robotec.ai>

adamdbrw commented on Gems/ROS2/Code/Source/Manipulator/PidMotorControllerComponent.cpp:
```diff
@@ -0,0 +1,89 @@
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &PidMotorControllerComponent::m_zeroOffset,
+                        "Zero Off.",
```
Zero Offset
(beter not to abbreviate for clarity)

adamdbrw reviewed (commented):
The change looks very good! This is definitely a big improvement!
I added some minor comments - please take a look. Also, please rebase to remove the conflicts.

alek-kam-robotec-ai commented on Gems/ROS2/Code/Source/Manipulator/JointMotorControllerComponent.cpp:
```diff
@@ -0,0 +1,109 @@
+            if (AZ::EditContext* ec = serializeContext->GetEditContext())
+            {
+                ec->Class<JointMotorControllerComponent>(
+                      "Joint Motor Controller Component", "Base component for motor controller components.")
```
The joint motor controller component is not supposed to be displayed as an addable component since it is only a base component class from which "addable" motor controllers derive. When it comes to other, usable controllers, they are already assigned a category.

alek-kam-robotec-ai commented:
> The change looks very good! This is definitely a big improvement! I added some minor comments - please take a look. Also, please rebase to remove the conflicts.

I have rebased the branch and adjusted the code to your review comments. Thanks!

michalpelka reviewed (approved):
Looks, a good note that some Roscon demo needs an update on that.
